pkg/controller/common/helpers.go

```diff
@@ -325,26 +326,56 @@ func convertIgnition32to31(ign3config ign3types.Config) (ign3_1types.Config, err
 func ValidateIgnition(ignconfig interface{}) error {
```

Comment:
I think it's worth considering putting this in `ValidateMachineConfig` per discussion here https://github.com/openshift/machine-config-operator/pull/2820#discussion_r745676939

I don't think @yuqi-zhang or I have strong opinions, but I do think it would be helpful to decide clearly where we're going to put validation, since this keeps coming up

Reply:
Also can you remind me how Ignition handles this? Are you imposing a restriction to what Ignition can handle? Or enforcing one of Ignitions restrictions?

Reply:
Hmm since this is an Ignition bug maybe leave the check in `ValidateIgnition` but add a comment to remove the check once https://github.com/coreos/ignition/issues/1301 gets fixed?

Reply:
It's better to make a jira card to track it than rely on in code comments.

Reply:
Agreed. I'll get one open and will edit this comment to include the card number.